src/kernels/kernelProvider.base.ts

```diff
@@ -53,6 +55,27 @@ export abstract class BaseKernelProvider implements IKernelProvider {
         disposables.push(this._onDidStartKernel);
         disposables.push(this._onDidCreateKernel);
     }
+    updateKernel(kernel: IKernel, metadata: KernelConnectionMetadata, controller: NotebookController): void {
+        const controllerWrapper = kernel.controller;
+        if (NotebookControllerWrapper.isWrapped(controllerWrapper)) {
+            controllerWrapper.update(controller);
+        } else {
+            throw new Error('IKernel instantiated without passing a NotebookControllerWrapper');
+        }
+        const metadataWrapper = kernel.kernelConnectionMetadata;
+        if (KernelConnectionMetadataProxy.isWrapped(metadataWrapper)) {
+            metadataWrapper.update(metadata);
+        } else {
+            throw new Error('IKernel instantiated without passing a KernelConnectionMetadataWrapper');
```

Comment:
Should this error message be localized?

Reply:
no, it's not meant for the user, these are not meant to happen. it's only for logging